### PR TITLE
[ACM-11392: Clarify storage support for HCP OCP Virtualization Provider

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-11392

### CODE REVIEW

File: clusters/hosted_control_planes/config_storage_kubevirt.adoc [hunk 1]
```diff
@@ -3,6 +3,29 @@
...
+
+| Any RWX `Block` CSI provider
+| Red Hat OpenShift Data Foundation external mode
+| Red Hat OpenShift Data Foundation feature set
+|
+
+| Any RWX `Block` CSI provider
+| Red Hat OpenShift Data Foundation internal mode
+| Red Hat OpenShift Data Foundation feature set
+| Do not use
```
Comment (zanetworker) [inline clusters/hosted_control_planes/config_storage_kubevirt.adoc:25]: Using ODF in the hosted cluster is not recommended. cc @davidvossel (can you take a look as well)

File: clusters/hosted_control_planes/config_storage_kubevirt.adoc [hunk 2]
```diff
@@ -3,6 +3,29 @@
...
 
+The following table illustrates the capabilities that the infrastructure must provide to support persistent storage in a hosted cluster:
+
+.Persistent storage modes in a hosted cluster
+|===
+| Infrastructure CSI provider | Hosted cluster CSI provider | Hosted cluster capabilities | Notes
+
+| Any RWX `Block` CSI provider
+| `kubevirt-csi`
+| Basic: RWO `Block` and `File` only
```
Comment (davidvossel) [inline clusters/hosted_control_planes/config_storage_kubevirt.adoc]: In 4.16, we introduced RWX Block and Snapshot support. So maybe this line should read

"RWO `Block` and `File`, RWX `Block`, Snapshots"

Comment (lahinson) [reply clusters/hosted_control_planes/config_storage_kubevirt.adoc]: Thanks! I updated the line with your suggestion.

File: clusters/hosted_control_planes/config_storage_kubevirt.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/config_storage_kubevirt.adoc:25]: Should I add "Do not use" to the row where ODF external mode is mentioned, too?

**Review by davidvossel — COMMENTED** [review]
/lgtm

File: clusters/hosted_control_planes/config_storage_kubevirt.adoc (hunk 1 above)
Comment (davidvossel) [reply clusters/hosted_control_planes/config_storage_kubevirt.adoc:25]: External mode is okay. It's mostly internal mode that causes performance issues.

**Review by jc-berger — APPROVED** [review]
/lgtm

**Pull request merged**